Sources/SwiftDocC/Model/DocumentationNode.swift

```diff
@@ -421,6 +422,41 @@ public struct DocumentationNode {
             semantic.httpResponsesSectionVariants[.fallback] = HTTPResponsesSection(responses: responses)
         }
         
+        if let possibleValues = markupModel.discussionTags?.possibleValues, !possibleValues.isEmpty {
+            let validator = PossibleValuesSection.Validator(diagnosticEngine: engine)
+            guard let symbolAllowedValues = symbol?.mixins[SymbolGraph.Symbol.AllowedValues.mixinKey] as? SymbolGraph.Symbol.AllowedValues else {
+                possibleValues.forEach { 
+                    engine.emit(validator.makeExtraPossibleValueProblem($0, knownPossibleValues: []))
+                }
+                return semantic.possibleValuesSectionVariants[.fallback] = PossibleValuesSection(possibleValues: [])
+            }
+            
+            // Drop possible values defined in the markdown that are not defined
+            // in the SymbolGraph and emit a warning.
+            var knownPossibleValues = possibleValues.filter { possibleValue in
+                if symbolAllowedValues.value.contains(where: { String($0) == possibleValue.value }) {
+                    return true
+                }
+                // Record problem about extra possible value.
+                engine.emit(
+                    validator.makeExtraPossibleValueProblem(possibleValue, knownPossibleValues: Set(symbolAllowedValues.value.map { String($0) }))
+                )
+                return false
+            }
```

Comment:
I find that the way that this both filters the values and does work on the filtered out values is a bit hard to read like this. I think it would be easier to follow what's happening if it was done in two distinct steps.

We have a `categorize(where:)` extension that could be used here to split the known and unknown values. Then a separate loop could emit diagnostics for the unknown values.

Also, by precomputing allowed value string values we can avoid a small inefficiency of recreating all the allowed value strings for each possible value.

```suggestion
            // Ignore documented possible values that don't exist in the symbol's allowed values in the symbol graph.
            let allowedPossibleValueNames = Set(symbolAllowedValues.value.map { String($0) })
            let (knownPossibleValues, unknownPossibleValues) = possibleValues.categorize(where: { allowedPossibleValueNames.contains($0.value) })
            for unknownValue in unknownPossibleValues {
                engine.emit(
                    validator.makeExtraPossibleValueProblem(unknownValue, knownPossibleValues: allowedPossibleValueNames)
                )
            }
```

Reply:
Additionally, with this being two separate steps. We could move the `for unknownValue in unknownPossibleValues` later in this function and use `knownPossibleValueNames` to avoid making suggestions for a diagnostic that the developer has already documented.